js/legend-i18n.js

```diff
@@ -1,186 +1,38 @@
-            "bust": "Buste",
-            "city-gate": "Porte de la ville",
-            "guidepost": "Panneau d'orientation",
-            "map": "Carte",
-            "memorial": "Mémorial",
-            "obelisk": "Obélisque",
-            "plaque": "Plaque",
-            "shrine": "Sanctuaire",
-            "statue": "Statue",
-            "stone": "Pierre",
-            "terminal": "Terminal",
-            "misc": "Divers",
-            "trafic-light": "Feux de signalisation",
-            "trafic-calming": "Dispositif pour réduire la vitesse",
-            "bike-zebra": "Passage zèbré pour bicyclettes",
-            "ford": "Gué",
-            "elevator": "Ascenseur (bleu si adapté aux vélos)",
-            "train-station": "Gare ferroviaire",
-            "aerodrome": "Aérodrome",
-            "helipad": "Héliport",
-            "bunker": "Bunker",
-            "chimney": "Cheminée (p.ex. usine)",
-            "crane": "Grue",
-            "cross": "Croix",
-            "fountain": "Fontaine",
-            "light-house": "Phare",
-            "storage-tank": "Réservoir de stockage",
-            "water-tower": "Château d'eau",
-            "wind-plant": "Centrale éolienne",
-            "wind-mill": "Moulin à vent",
-            "barriers": "Barrières",
-            "cattle-grid": "Grille pour bétail",
-            "cycle-barrier": "Barrière ralentissant ou interdisant les vélos",
-            "gate": "Portail",
-            "kissing-gate": "Portillon à chicanes",
-            "lift-gate": "Barrière levante",
-            "stile": "Échalier",
-            "blocks": "Borne, bloc, muret de sécurité"
-        }
+    const file = `i18n/${locale}.json`;
+    const response = await fetch(file);
+    if (!response.ok) {
+        console.error(`Failed to load localization file: ${file}`);
+        return;
     }
-    banana.load(messages, banana.locale)
+    const data = await response.json();
+    banana.load(data, locale);
+    banana.setLocale(locale);
+    console.log(`Localization loaded for locale: ${locale}`);
     document.querySelectorAll('.i18n').forEach(element => {
-        let message = element.dataset.i18n
+        const message = element.dataset.i18n;
         if (message === undefined) {
-            return
+            return;
         }
-        let localizedMessage = banana.i18n(message)
-        if (message !== undefined && message != localizedMessage) {
-            element.innerText = banana.i18n(message)
+        const localizedMessage = banana.i18n(message);
+        if (message !== localizedMessage) {
+            element.innerText = localizedMessage;
         }
-    })
-})
+    });
+}
+
+window.addEventListener('load', () => {
+    const userLang = navigator.language || navigator.userLanguage;
+    let locale = 'en';
+    if (userLang.startsWith('fr')) {
+        locale = 'fr';
+    } else if (userLang.startsWith('es')) {
+        locale = 'es';
+    }
+    loadLocalization(locale);
+});
```

Comment:
```suggestion
window.addEventListener('load', () => {
    const userLang = navigator.language || navigator.userLanguage;
    loadLocalization(userLang);
});
```
You could define the function as:
```
async function loadLocalization(locale = 'en') {
```
In case, your browser was set in german, for instance, the `fetch` clause would fail, returning the english by default.

In this way, you won't need to edit the code whenever a new language is added

Reply:
Great suggestion, thanks! I had to update the code to be able to handle region codes like `es-ES` and fallback to `es`. Let me know what you think!

Reply:
you could simplify the full logic like:
```
async function loadTranslation(languages = navigator.languages) {
  for (const lang of languages) {
    try {
      const response = await fetch(`/locales/${lang}.json`);
      if (response.ok) {
        const data = await response.json();
        console.info(`Traducción cargada: ${lang}`);
        return data;
      }
    } catch (e) {
      // Puede fallar el fetch si el archivo no existe: ignoramos y seguimos
      console.warn(`No se encontró /locales/${lang}.json`);
    }
  }

  // Si no se encuentra ninguna, usar el fallback
  console.warn("Usando traducción por defecto en inglés.");
  return;
}
```

Reply:
Let's say your browser languages are `['es-MX','es']`. The first attempt will try to load _es-MX_, if it does, the function exits and it's done. If not, it continues to _es_, and so on.

Even you could add a fast return (before the try) as `if (lang === "en") return`, for english browsers to avoid any useless XHR request.

Reply:
@Crashillo, indeed, your approach looks cleaner. I have implemented it with a few changes. Let me know what you think!